docs/proposals/21-ipv6-singlestack-local.md

```diff
@@ -0,0 +1,213 @@
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+  * [Proposal](#proposal)
+    * [`Shoot` API](#shoot-api)
+    * [`Seed` API](#seed-api)
+    * [`DNSRecord` API](#dnsrecord-api)
+    * [Implementation Overview](#implementation-overview)
+      * [Preparing the Local Setup](#preparing-the-local-setup)
+      * [Scheduling](#scheduling)
+      * [DNS Records](#dns-records)
+      * [Network Policies](#network-policies)
+      * [Shoot Worker Node Kernel Configuration](#shoot-worker-node-kernel-configuration)
+      * [Docker Hub Images](#docker-hub-images)
+      * [E2E Tests](#e2e-tests)
+      * [Networking Extensions](#networking-extensions)
+  * [Alternatives Considered](#alternatives-considered)
+<!-- TOC -->
+
+## Summary
+
+Today, all shoot clusters provisioned by Gardener use IPv4 single-stack networking.
+This GEP proposes changes to support IPv6 single-stack networking for shoots using provider-local.
+
+## Motivation
+
+Depending on the underlying cloud infrastructure, there is a need to provide shoot networking setups different from what Gardener currently provides.
+This might include IPv6 single-stack or IPv4/IPv6 dual-stack setups.
+Kubernetes already supports such setups as described in [this doc](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).
+However, supporting these in Gardener requires changes to the API and many components (networking extensions, VPN tunnel, etc.).
+
+To bring Gardener closer to supporting these kinds of networking setups, this GEP focuses on supporting IPv6 single-stack networking in the local Gardener environment.
+This keeps things focused on the most important changes to the API and central components while neglecting infrastructure-specific quirks and other difficulties that will arise with dual-stack networking.
+
+Once this enhancement has been implemented and IPv6 single-stack networking in local shoots is a stable feature, further changes can be proposed to support IPv6 single-stack on cloud infrastructure or IPv4/IPv6 dual-stack networking.
+
+### Goals
+
+- augment the Gardener API to allow selecting either IPv4 or IPv6 single-stack networking
+  - define a contract that all components (including networking extensions) need to follow to support this configuration
+  - allow extending the API for dual-stack networking later on while keeping backward-/forward-compatibility
+- adapt all relevant components in the local Gardener environment to support this configuration
+- add e2e tests to prevent regressions and guarantee the stability of the feature (especially because nobody is running it in productive environments)
+
+### Non-Goals
+
+- support IPv4/IPv6 dual-stack networking
+- support IPv6 single-stack networking on cloud infrastructure
+- support changing the networking setup of shoots from IPv4 to IPv6 single-stack or vice-versa
+- propose changes to `NetworkConfig` APIs of networking extensions or their implementation
```

Comment:
> Do I understand you correctly that you do want to change the `Network` custom resource, but not the `NetworkConfig` sub-resource?

Yes, exactly. We only want to change the API types defined in this repo (including `Network`). But we explicitly do not want to dictate changes to the out-of-tree APIs (`NetworkConfig`).

You're right, this is confusing because we missed mentioning the `Network` API. I added it accordingly.